src/trackinfoobject.cpp

```diff
@@ -566,46 +562,65 @@ void TrackInfoObject::setTrackNumber(const QString& s) {
     }
 }
 
-int TrackInfoObject::getTimesPlayed() const {
+void TrackInfoObject::setPlayed(bool played) {
     QMutexLocker lock(&m_qMutex);
-    return m_iTimesPlayed;
+    PlayCounter playCounter(m_playCounter);
+    playCounter.setPlayed(played);
+    if (m_playCounter != playCounter) {
+        m_playCounter = playCounter;
+        setDirty(true);
+    }
 }
 
-void TrackInfoObject::setTimesPlayed(int t) {
+bool TrackInfoObject::isPlayed() const {
     QMutexLocker lock(&m_qMutex);
-    if (t != m_iTimesPlayed) {
-        m_iTimesPlayed = t;
+    return m_playCounter.isPlayed();
+}
+
+void TrackInfoObject::setTimesPlayed(int timesPlayed) {
+    QMutexLocker lock(&m_qMutex);
+    PlayCounter playCounter(m_playCounter);
+    playCounter.setTimesPlayed(timesPlayed);
+    if (m_playCounter != playCounter) {
+        m_playCounter = playCounter;
         setDirty(true);
     }
 }
 
-void TrackInfoObject::incTimesPlayed() {
-    setPlayedAndUpdatePlaycount(true);
+void TrackInfoObject::resetTimesPlayed() {
+    QMutexLocker lock(&m_qMutex);
+    PlayCounter playCounter(m_playCounter);
+    playCounter.resetTimesPlayed();
```

Comment:
As I mentioned both play counter and the played flag are relevant for dirty!

ut I will remove all functions from TrackInfoObject that modify only 
individual attributes of the PlayCounter.

On 11/12/2015 09:35 PM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.cpp 
> <https://github.com/mixxxdj/mixxx/pull/772#discussion_r44710686>:
>
> >          setDirty(true);
> >      }
> >  }
> >
> > -void TrackInfoObject::incTimesPlayed() {
> > -    setPlayedAndUpdatePlaycount(true);
> > +void TrackInfoObject::resetTimesPlayed() {
> > +    QMutexLocker lock(&m_qMutex);
> > +    PlayCounter playCounter(m_playCounter);
> > +    playCounter.resetTimesPlayed();
>
> Maybe we should get rid of resetTimesPlayed() and here. It is confusing 
> since the played counter is not relevant for "dirty".
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/772/files#r44710686>.
>